core/HoistBase.js

```diff
@@ -0,0 +1,275 @@
+    }
+
+    /**
+     * Load this object. Implement this method to describe how this object should load
+     * itself from underlying data sources or services.
+     *
+     * For implementation only.  Callers should call loadAsync() or refreshAsync() instead.
+     *
+     * @param {LoadSpec} loadSpec - Metadata about the underlying request. Implementations should
+     *      take care to pass this parameter to any delegates (e.g. other LoadSupport instances)
+     *      that accept it.
+     */
+    async doLoadAsync(loadSpec) {}
+
+
+    //-------------------
+    // Other
+    //-------------------
+
+    /**
+     *  Method to make a class property persistent, syncing its value via a configured
+     * `PersistenceProvider` to maintain and restore values across browser sessions.
+     *
+     * This may be used on any @observable or @bindable class property which is a primitive.
+     * It will initialize the observable's value from the class's default
+     * `PersistenceProvider` and will subsequently write back any changes to the property
+     * to that Provider. If the Provider has not yet been populated with a value, or an
+     * error occurs, it will use the value set in-code instead.
+     *
+     * @param {string} property - name of property on this object to bind to a
+     *      persistence provider.
+     * @param {PersistOptions} [options] - options governing the persistence of this
+     *      object.  These will be applied on top of any default persistWith options defined
+     *      on the instance itself.
+     *
+     * See also @persist and @persist.with for a convenient decorator that may be used
+     * directly on the property declaration itself.  Use this method in the general case,
+     * when you need to control the timing.
+     */
+    markPersist(property, options = {}) {
+        // Read from and attach to Provider, failing gently
+        try {
+            const persistWith = {path: property, ...this.persistWith, ...options},
+                provider = this.markManaged(PersistenceProvider.create(persistWith)),
+                providerState = provider.read();
+            if (!isUndefined(providerState)) {
+                runInAction(() => this[property] = cloneDeep(providerState));
+            }
+            this.addReaction({
+                track: () => this[property],
+                run: (data) => provider.write(data)
+            });
+        } catch (e) {
+            console.error(
+                `Failed to configure Persistence for '${property}'.  Be sure to fully specify ` +
+                `'persistWith' on this object or in the method call.`
+            );
+        }
+    }
+
+
+    /**
+     * Clean up resources associated with this object
+     */
+    destroy() {
+        this._xhManagedProperties?.forEach(p => XH.safeDestroy(this[p]));
+        this._xhManagedInstances?.forEach(i => XH.safeDestroy(i));
+        XH.safeDestroy(this._loadModel);
```

Comment:
Any preference for marking this as managed above, where it's constructed, vs. explicitly destroying it here? Do we want to encourage one over the other as a general best practice?

Reply:
yes -- will mark it as managed.  In an earlier iteration, I think it was not possible, but now it makes total sense.  At the very least, it provides a good example of how HoistBase and @managed is supposed to work.